packages/modules/kostal_plenticore/device.py

```diff
@@ -0,0 +1,165 @@
+                        # in der PV-Leistung ersichtlich.
+                        pv_state = InverterState(power=inverter_state.power + bat_state_net.power,
+                                                 exported=inverter_state.exported)
+                        bat_state_gross = bat_state_net
+                    # https://github.com/snaptec/openWB/pull/2440#discussion_r996275286
+                    # power_gross = bat_state.power + dc_in
+                    # bat_state_gross = BatteryState(power=bat_state_net.power / power_gross * inverter_state.power)
+                    # pv_state = InverterState(power=inverter_state.power - bat_state_gross.power)
+                else:
+                    pv_state = inverter_state
+                    bat_state_gross = bat_state_net
+            else:
+                pv_state = inverter_state
+            if set_inverter_state:
+                component.set(pv_state)
+        elif isinstance(component, KostalPlenticoreCounter):
+            component.update(reader)
+    if bat_state_net:
+        battery.set(bat_state_gross)
+    if set_inverter_state is False:
+        return pv_state
+
+
+def create_device(device_config: KostalPlenticore):
+    def create_bat_component(component_config):
+        return KostalPlenticoreBat(device_config.id, component_config)
+
+    def create_counter_component(component_config):
+        return KostalPlenticoreCounter(device_config.id, component_config)
+
+    def create_inverter_component(component_config):
+        return KostalPlenticoreInverter(component_config)
+
+    def update_components(
+        components: Iterable[Union[KostalPlenticoreBat, KostalPlenticoreCounter, KostalPlenticoreInverter]]
+    ):
+        with tcp_client:
+            update(components, reader)
+
+    tcp_client = modbus.ModbusTcpClient_(device_config.configuration.ip_address, 1502)
+    reader = _create_reader(tcp_client)
+    return ConfigurableDevice(
+        device_config=device_config,
+        component_factory=ComponentFactoryByType(
+            bat=create_bat_component, counter=create_counter_component, inverter=create_inverter_component),
+        component_updater=MultiComponentUpdater(update_components),
+    )
+
+
+def _create_reader(tcp_client: modbus.ModbusTcpClient_) -> Callable[[int, modbus.ModbusDataType], Any]:
+    return functools.partial(tcp_client.read_holding_registers, unit=71, wordorder=Endian.Little)
+
+
+def read_legacy_inverter(ip1: str, ip2: str, battery: int, ip3: str) -> InverterState:
+    # in IP3 kann ein aufeinanderfolgende Liste enthalten sein "192.168.0.1-3"
+    log.debug("Kostal Plenticore: WR1: %s, WR2: %s, WR3: %s, Battery: %s", ip1, ip2, ip3, battery)
+    inverter_component = KostalPlenticoreInverter(KostalPlenticoreInverterSetup(id=1))
+
+    def get_hybrid_inverter_state(ip: str) -> InverterState:
+        battery_component = KostalPlenticoreBat(1, KostalPlenticoreBatSetup())
+        with modbus.ModbusTcpClient_(ip) as client:
```

Comment:
Ich war in meiner letzten Runde da etwas schnell. Mir ist noch aufgefallen, dass du vorher port 1502 mitgegeben hast. Das sollte vermutlich erhalten bleiben.